python-lib/xlsx_writer.py

```diff
@@ -104,33 +106,94 @@ def auto_size_column_width(worksheet: Worksheet):
+        - truncate the name to 28 chars
+        - Add an index from 00 to 99 at the end in case of overlap
+    :param input_dataset_names: the list of dataset names to remap
+    :returns: a Dict[str, str] mapping the DS names with the sheet names
     """
-    logger.info("Writing output xlsx file ...")
-    writer = pd.ExcelWriter(xlsx_abs_path, engine='openpyxl')
 
-    for name in input_dataframes_names:
-        df = dataframe_provider(name)
+    return_map = {}
+    index_rename=-1
+    renaming_length = EXCEL_MAX_LEN_SHEET_NAME-2
+    for name in input_dataset_names:
+        if len(name) > EXCEL_MAX_LEN_SHEET_NAME:
+            index_rename+=1
+            rename= f"{name[0:renaming_length]}{index_rename:02d}"
+            # Almost impossible case : a DS already has this name
+            while rename in input_dataset_names:
+                index_rename+=1
+                rename= f"{name[0:renaming_length]}{index_rename:02d}"
 
-        logger.info("Writing dataset into excel sheet...")
-        df.to_excel(writer, sheet_name=name, index=False, encoding='utf-8')
+            logger.info(f"Dataset {name} with a too long name will be stored as sheet {rename}")
+            return_map[name]=rename
+        else:
+            return_map[name]=name
 
-        worksheet = writer.sheets.get(name)
+    return return_map
 
-        if worksheet is None:
-            logger.warn(f"No worksheet for dataset {name}. Written but styling skipped.")
-            continue
 
-        logger.info(f"Styling excel sheet...")
-        style_header(worksheet)
-        auto_size_column_width(worksheet)
+def datasets_to_xlsx(input_dataset_names, xlsx_abs_path, worksheet_provider):
+    """
+    Write the input datasets into same excel into the folder
+    :param input_dataset_names: the list of dataset to put in a single excel file, using one sheet (excel tab) per dataset
+    :param xlsx_abs_path: the temporary path where to write the final excel file
+    :param dataset_provider: a lambda used to get the dataset
+    """
 
-        logger.info("Finished writing dataset {} into excel sheet.".format(name))
+    logger.info(f"Building output excel file ... {xlsx_abs_path}")
+    # The final workbook where all dataset sheets will be written
+    workbook = Workbook()
+    # remove the default sheet created
+    workbook.remove(workbook.active)
+
+    renaming_map = rename_too_long_dataset_names(input_dataset_names)
+
+    for name in input_dataset_names:
+        ds_worksheet = worksheet_provider(name)
+        if ds_worksheet is None:
+            continue
 
-    writer.save()
+        if name in renaming_map:
+            ds_worksheet.title = renaming_map[name]
+        else:
+            # should never happen
+            ds_worksheet.title = name
+        
+        target_sheet = copy_sheet_to_workbook(ds_worksheet, workbook)
+        
+        logger.info(f"Styling excel sheet {target_sheet.title} in target workbook")
+        style_header(target_sheet)
```

Comment:
I believe this is no longer useful as the export already styles the header (color and font change). The entire function could be removed with the associated constants, except maybe if we have in the roadmap to give header personalisation options
```suggestion
```

Reply:
done